libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -165,30 +170,32 @@ wasm::Expression WasmCodeTransform::operator()(Identifier const& _identifier)
 
 wasm::Expression WasmCodeTransform::operator()(Literal const& _literal)
 {
-	u256 value = valueOfLiteral(_literal);
-	yulAssert(value <= numeric_limits<uint64_t>::max(), "Literal too large: " + value.str());
-	return wasm::Literal{static_cast<uint64_t>(value)};
+	return makeLiteral(translatedType(_literal.type), valueOfLiteral(_literal));
 }
 
 wasm::Expression WasmCodeTransform::operator()(If const& _if)
 {
-	// TODO converting i64 to i32 might not always be needed.
+	yul::Type conditionType = m_typeInfo.typeOf(*_if.condition);
+	yulAssert(m_dialect.types.count(conditionType) == 1, "");
```

Comment:
Instead of this, you could check if `conditionType.str() + ".ne"` is a valid builtin function.

Reply:
Good point. I'll do that.

Reply:
Done.